java8/src/main/java/io/cucumber/java8/Java8DataTableTypeDefinition.java

```diff
@@ -0,0 +1,48 @@
+package io.cucumber.java8;
+
+import io.cucumber.core.backend.DataTableTypeDefinition;
+import io.cucumber.datatable.DataTableType;
+import io.cucumber.datatable.TableEntryTransformer;
+import io.cucumber.datatable.TableTransformer;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Map;
+import net.jodah.typetools.TypeResolver;
+
+final class Java8DataTableTypeDefinition extends AbstractGlueDefinition implements DataTableTypeDefinition {
+    
+    private final DataTableType dataTableType;
+    
+    Java8DataTableTypeDefinition(Object body) {
+        super(body, new Exception().getStackTrace()[3]);
+        this.dataTableType = createDataTableType(method);
+    }
+
+    private DataTableType createDataTableType(Method method) {
+        Class returnType = TypeResolver.resolveRawArguments(DataTableDefinitionBody.class, body.getClass())[0];
+        Type[] parameterTypes = method.getGenericParameterTypes();
+        Type parameterType = parameterTypes[0];
+
+        if (Map.class.equals(parameterType)) {
+            return new DataTableType(
+                returnType,
+                (TableEntryTransformer<Object>) this::execute
+            );
+        }
+        else {
+            return new DataTableType(
+                returnType,
+                (TableTransformer<Object>) this::execute
```

Comment:
[Coveralls indicates there is no coverage for this branch](https://coveralls.io/builds/25987334/source?filename=java8/src/main/java/io/cucumber/java8/Java8DataTableTypeDefinition.java#L33). And looking at [`JavaDataTableTypeDefinition.createDataTableType](https://github.com/cucumber/cucumber-jvm/blob/master/java/src/main/java/io/cucumber/java/JavaDataTableTypeDefinition.java#L71) I think there are some options missing.